packages/wdio-cucumber-framework/src/index.ts

```diff
@@ -253,58 +240,68 @@ class CucumberAdapter {
     async run() {
         let runtimeError
         let result
+        let failedCount
+
         try {
             await this.registerRequiredModules()
             supportCodeLibraryBuilder.reset(this._cwd, this._newId)
 
-            /**
-             * wdio hooks should be added before spec files are loaded
-             */
-            this.addWdioHooks(this._config)
-            await this.loadSpecFiles()
-            this.wrapSteps(this._config)
+            this.addWdioHooksAndWrapSteps(this._config, supportCodeLibraryBuilder)
 
-            /**
-             * we need to somehow identify is function is step or hook
-             * so we wrap every user hook function
-             */
-            setUserHookNames(supportCodeLibraryBuilder)
             setDefaultTimeout(this._cucumberOpts.timeout)
+
+            await this.loadFiles()
+
             const supportCodeLibrary = supportCodeLibraryBuilder.finalize()
 
-            /**
-             * gets current step data: `{ uri, feature, scenario, step, sourceLocation }`
-             * or `null` for some hooks.
-             */
-            this.getHookParams = this._cucumberReporter
-                .eventListener
-                .getHookParams
-                .bind(this._cucumberReporter.eventListener)
-
-            const runtime = new Runtime({
-                newId: this._newId,
-                eventBroadcaster: this._eventBroadcaster,
-                options: this._cucumberOpts as any as IRuntimeOptions,
-                supportCodeLibrary,
-                eventDataCollector: this._eventDataCollector,
-                pickleIds: this._cucumberReporter!.eventListener.getPickleIds(this._capabilities)
+            const outStream = new Writable({
+                write(chunk, encoding, callback) {
+                    callback()
+                },
+            })
```

Comment:
Are we using this stream anywhere else? Does it stream some important information?

Reply:
The purpose of this stream is to actually suppress the standard output returned from cucumber-js. In real-world scenarios, this behaviour will be taken over by the `cucumberFormatter`. However, during unit testing, we are suppressing the output intentionally.

Reply:
Gotcha, let's make sure we close the stream at the end so we don't have any pending listeners hanging around.